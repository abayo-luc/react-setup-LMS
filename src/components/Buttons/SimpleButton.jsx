import React from 'react';
import PropTypes from 'prop-types';

const SimpleButton = (props) => {
  const { title, onClick, disabled } = props;
  return (
    <button
      type="button"
      className="btn btn-primary btn-lg btn-block"
      onClick={onClick}
      disabled={disabled}
    >
      {title}
    </button>
  );
};
SimpleButton.propTypes = {
  title: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
};
SimpleButton.defaultProps = {
  onClick: () => null,
  disabled: false,
};
export default SimpleButton;
