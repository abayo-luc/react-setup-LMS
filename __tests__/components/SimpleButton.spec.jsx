import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SimpleButton from '../../src/components/Buttons/SimpleButton';

const warper = shallow(<SimpleButton title="Submit" />);
describe('Simple Button', () => {
  test('should match the snapshot', () => {
    expect(toJson(warper)).toMatchSnapshot();
  });
});
