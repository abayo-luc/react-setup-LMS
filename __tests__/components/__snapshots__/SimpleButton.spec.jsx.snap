// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple Button should match the snapshot 1`] = `
<button
  className="btn btn-primary btn-lg btn-block"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Submit
</button>
`;
